code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -164,8 +164,19 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 	if(user.canUseTopic(src, be_close = TRUE))
 		var/obj/item/computer_hardware/card_slot/card_slot2 = all_components[MC_CARD2]
 		var/obj/item/computer_hardware/card_slot/card_slot = all_components[MC_CARD]
-		if(card_slot2?.try_eject(user) || card_slot?.try_eject(user))
+
+		if(istype(card_slot) && card_slot.stored_card && card_slot?.try_eject(user))
+			return TRUE
+
+		if(istype(card_slot2) && card_slot2?.stored_card && card_slot2?.try_eject(user))
+			return TRUE
+
+		if(istype(inserted_pai)) // Remove pAI
+			user.put_in_hands(inserted_pai)
+			balloon_alert(user, "removed pAI")
+			inserted_pai = null
```

Comment:
Move this to exited(), that way it's cleared automatically during qdel and if admin uses the get command